lib/util/esWrapper.js

```diff
@@ -114,63 +121,106 @@ class ESWrapper {
     }
 
     if (error instanceof es.errors.NoConnections) {
-      return new ServiceUnavailableError('Elasticsearch service is not connected');
+      errorsManager.throw(
+        'external',
+        'elasticsearch',
+        'elasticsearch_service_not_connected');
     }
 
     let message = error.message || '';
 
-    const messageReplaced = errorMessagesMapping.some(mapping => {
-      message = message.replace(mapping.regex, mapping.replacement);
-      return message !== error.message;
-    });
+    let kuzzleError = null;
+
+    for (const mapping of errorMessagesMapping) {
+      const matches = message.match(mapping.regex, mapping.replacement);
+
+      if (!matches) {
+        continue;
+      }
 
-    let kuzzleError;
+      kuzzleError = errorsManager.getError(
+        'external',
+        'elasticsearch',
+        mapping.subcode,
+        ...mapping.getPlaceholders(matches));
+
+      break;
+    }
 
     switch (error.displayName) {
       case 'BadRequest':
-        if (!messageReplaced) {
-          if (error.body && error.body.error) {
-            message = error.body.error.root_cause ?
-              error.body.error.root_cause[0].reason :
-              error.body.error.reason;
-          }
+        if (kuzzleError) {
+          break;
+        }
 
-          debug('unhandled "BadRequest" elasticsearch error: %a', error);
+        if (error.body && error.body.error) {
+          message = error.body.error.root_cause
+            ? error.body.error.root_cause[0].reason
+            : error.body.error.reason;
         }
 
-        kuzzleError = new BadRequestError(error);
+        kuzzleError = errorsManager.getError(
+          'external',
+          'elasticsearch',
+          'unexpected_bad_request_error',
+          message);
+
+        debug('unhandled "BadRequest" elasticsearch error: %a', error);
+
         break;
+
       case 'NotFound':
-        if (!messageReplaced) {
-          if (error.body && error.body.error) {
-            message = error.body.error
-              ? error.body.error.reason + ': ' + error.body.error['resource.id']
-              : error.message + ': ' + error.body._id;
-          }
+        if (kuzzleError) {
+          break;
+        }
 
-          debug('unhandled "NotFound" elasticsearch error: %a', error);
+        if (error.body && error.body.error) {
+          message = error.body.error
+            ? `${error.body.error.reason}: ${error.body.error['resource.id']}`
+            : `${error.message}: ${error.body._id}`;
         }
 
-        kuzzleError = new NotFoundError(error);
+        kuzzleError = errorsManager.getError(
+          'external',
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "message" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1854)